sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2120,6 +2120,19 @@ class Dataset[T] private[sql](
   }
 
   /**
+   * Returns a new Dataset by adding columns with metadata.
+   */
+  private[spark] def withColumns(
```

Comment:
This is not being tested in SQL

Reply:
Yeah, I see. I've left comment https://github.com/apache/spark/pull/17819#discussion_r142172037 that I will add test later.